Content.Server/Ninja/Systems/StunProviderSystem.cs

```diff
@@ -55,8 +60,12 @@ private void OnBeforeInteractHand(EntityUid uid, StunProviderComponent comp, Bef
 
         _audio.PlayPvs(comp.Sound, target);
 
-        // not holding hands with target so insuls don't matter
-        _electrocution.TryDoElectrocution(target, uid, comp.StunDamage, comp.StunTime, false, ignoreInsulation: true);
+        // do the shock
+        var shock = _proto.Index<DamageTypePrototype>("Shock");
+        var damage = new DamageSpecifier(shock, comp.StunDamage);
+        _damageable.TryChangeDamage(target, damage, false, true, null, origin: uid);
```

Comment:
just put a damagespecifier on stunprovidercomponent instead of manually creating one

Reply:
yes thats what i did first but the damagespecifier constructor is evil
`new DamageSpecifier(IoCManager.Resolve<IPrototypeManager>().Index<DamageTypePrototype>("Shock"), 5)`

should i really do this or add a sane constructor for hardcoded prototype ids since it just uses .ID on the prototype nothing else

Reply:
also this could be avoided in general if there was a good way to copy a set of components from entity A to B, so gloves has StunProvider set up in its yaml then everything is copied properly so any yaml changes in the glove get reflected and dont have to rely on defaults...

Reply:
instead of adding and removing the component all the time, just add an "enabled" boolean on all the ninja glove comps and define all this shit in yaml.

Reply:
if i define it in yaml how do i copy the components to the user

Reply:
this is a big undertaking and wouldnt just affect stun provider every ability would use it

Reply:
I think just using the ctor without the validation is fine for now.